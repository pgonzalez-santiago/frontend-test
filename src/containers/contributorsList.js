import React, { Fragment, PureComponent } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

// Redux
// Redux
import RepositoryContributorsActions from '../store/reducers/repositoryContributors'

// Components
import Spinner from '../components/spinner'
import ContibutorItem from '../components/contributorItem'

// External dependencies
import styled from 'styled-components'
import Colors from '../assets/colors'

// Languages
import I18n from '../assets/lang'

const Title = styled.div`
  border-bottom: 1px solid ${Colors.primaryColor};
  padding-bottom: 5px;
  text-transform: capitalize;
`

const ListWrapper = styled.div`
  padding-top: 20px;
`

const MessageWrapper = styled.div`
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: center;
  color: ${Colors.primaryColor};
  text-transform: uppercase;
  padding: 20px;
`

export class ContributorsList extends PureComponent {
  componentDidMount () {
    const { getContributors, repoName } = this.props

    if (repoName) {
      // Get contributors when component is mounted
      getContributors('facebook', repoName)
    }
  }

  componentDidUpdate (prevProps) {
    const { getContributors, repoName } = this.props

    // If repository changes get the new contributors
    if (repoName !== prevProps.repoName) {
      getContributors('facebook', repoName)
    }
  }

  renderContent () {
    const { contributors, error, fetching } = this.props

    // If something goes wrong in the request render a message
    if (error) {
      return (
        <MessageWrapper>
          <span>{I18n.t('errorMessage')}</span>
        </MessageWrapper>
      )
    }

    // Render a spinner while contributors data is being fetched
    if (fetching) {
      return (
        <MessageWrapper>
          <Spinner
            color={Colors.primaryColor}
            size={30}/>
        </MessageWrapper>
      )
    }

    // Render contributors
    return (
      <ListWrapper>
        {contributors.map((item, index) => {
          return (
            <ContibutorItem
              key={item.id}
              contributor={item}
            />
          )
        })}
      </ListWrapper>
    )
  }

  render () {
    return (
      <Fragment>
        <Title>
          {I18n.t('contributorsList')}
        </Title>
        {this.renderContent()}
      </Fragment>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    contributors: state.repositoryContributors.contributors,
    error:        state.repositoryContributors.error,
    fetching:     state.repositoryContributors.fetching,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    getContributors: (repoName, owner) => dispatch(RepositoryContributorsActions.request(repoName, owner)),
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ContributorsList)

ContributorsList.propTypes = {
  contributors:    PropTypes.array,
  error:           PropTypes.bool.isRequired,
  fetching:        PropTypes.bool.isRequired,
  getContributors: PropTypes.func.isRequired,
  repoName:        PropTypes.string,
}
