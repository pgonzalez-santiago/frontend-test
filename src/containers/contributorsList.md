# Contributors List

----
## Props

* **repoName**             String: selected repo.

----
## Usage

```html
<ContributorsList repoName={repoName} />
```
