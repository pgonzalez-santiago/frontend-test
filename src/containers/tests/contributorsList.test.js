import React from 'react'
import { ContributorsList } from '../contributorsList'
import { shallow } from 'enzyme'

describe('ContributorsList', () => {
  it('ContributorsList Dom structure has not change when error', () => {
    const wrapper = shallow(
      <ContributorsList
        contributors={[]}
        error={true}
        fetching={false}
        getContributors={() => null}
        repoName={'react'}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('ContributorsList Dom structure has not change when fetching', () => {
    const wrapper = shallow(
      <ContributorsList
        contributors={[]}
        error={false}
        fetching={true}
        getContributors={() => null}
        repoName={'react'}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('ContributorsList Dom structure has not change when render contributors', () => {
    const wrapper = shallow(
      <ContributorsList
        contributors={[
          {
            id: 1,
            avatar_url:    'https://avatars3.githubusercontent.com/u/22032?v=4',
            contributions: 623,
            html_url:      'https://github.com/shergin',
            login:         'shergin',
          },
          {
            id: 2,
            avatar_url:    'https://avatars3.githubusercontent.com/u/5676?v=4',
            contributions: 510,
            html_url:      'https://github.com/javache',
            login:         'javache',
          }
        ]}
        error={false}
        fetching={false}
        getContributors={() => null}
        repoName={'react'}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
