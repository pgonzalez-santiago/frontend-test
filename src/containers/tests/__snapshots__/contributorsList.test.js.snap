// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContributorsList ContributorsList Dom structure has not change when error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContributorsList
    contributors={Array []}
    error={true}
    fetching={false}
    getContributors={[Function]}
    repoName="react"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          Contributors list
        </ForwardRef>,
        <ForwardRef>
          <span>
            Ups something went wrong!!
          </span>
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Contributors list",
        },
        "ref": null,
        "rendered": "Contributors list",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": true,
            "rules": Array [
              "
  border-bottom: 1px solid ",
              "#b8b7ad",
              ";
  padding-bottom: 5px;
  text-transform: capitalize;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <span>
            Ups something went wrong!!
          </span>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Ups something went wrong!!",
          },
          "ref": null,
          "rendered": "Ups something went wrong!!",
          "type": "span",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: center;
  color: ",
              "#b8b7ad",
              ";
  text-transform: uppercase;
  padding: 20px;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Contributors list
          </ForwardRef>,
          <ForwardRef>
            <span>
              Ups something went wrong!!
            </span>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Contributors list",
          },
          "ref": null,
          "rendered": "Contributors list",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": true,
              "rules": Array [
                "
  border-bottom: 1px solid ",
                "#b8b7ad",
                ";
  padding-bottom: 5px;
  text-transform: capitalize;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <span>
              Ups something went wrong!!
            </span>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Ups something went wrong!!",
            },
            "ref": null,
            "rendered": "Ups something went wrong!!",
            "type": "span",
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-EHOje",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: center;
  color: ",
                "#b8b7ad",
                ";
  text-transform: uppercase;
  padding: 20px;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-EHOje",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ContributorsList ContributorsList Dom structure has not change when fetching 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContributorsList
    contributors={Array []}
    error={false}
    fetching={true}
    getContributors={[Function]}
    repoName="react"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          Contributors list
        </ForwardRef>,
        <ForwardRef>
          <Spinner
            color="#b8b7ad"
            size={30}
          />
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Contributors list",
        },
        "ref": null,
        "rendered": "Contributors list",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": true,
            "rules": Array [
              "
  border-bottom: 1px solid ",
              "#b8b7ad",
              ";
  padding-bottom: 5px;
  text-transform: capitalize;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Spinner
            color="#b8b7ad"
            size={30}
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "#b8b7ad",
            "size": 30,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: center;
  color: ",
              "#b8b7ad",
              ";
  text-transform: uppercase;
  padding: 20px;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Contributors list
          </ForwardRef>,
          <ForwardRef>
            <Spinner
              color="#b8b7ad"
              size={30}
            />
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Contributors list",
          },
          "ref": null,
          "rendered": "Contributors list",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": true,
              "rules": Array [
                "
  border-bottom: 1px solid ",
                "#b8b7ad",
                ";
  padding-bottom: 5px;
  text-transform: capitalize;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Spinner
              color="#b8b7ad"
              size={30}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "color": "#b8b7ad",
              "size": 30,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-EHOje",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: center;
  color: ",
                "#b8b7ad",
                ";
  text-transform: uppercase;
  padding: 20px;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-EHOje",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`ContributorsList ContributorsList Dom structure has not change when render contributors 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ContributorsList
    contributors={
      Array [
        Object {
          "avatar_url": "https://avatars3.githubusercontent.com/u/22032?v=4",
          "contributions": 623,
          "html_url": "https://github.com/shergin",
          "id": 1,
          "login": "shergin",
        },
        Object {
          "avatar_url": "https://avatars3.githubusercontent.com/u/5676?v=4",
          "contributions": 510,
          "html_url": "https://github.com/javache",
          "id": 2,
          "login": "javache",
        },
      ]
    }
    error={false}
    fetching={false}
    getContributors={[Function]}
    repoName="react"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          Contributors list
        </ForwardRef>,
        <ForwardRef>
          <ContibutorItem
            contributor={
              Object {
                "avatar_url": "https://avatars3.githubusercontent.com/u/22032?v=4",
                "contributions": 623,
                "html_url": "https://github.com/shergin",
                "id": 1,
                "login": "shergin",
              }
            }
          />
          <ContibutorItem
            contributor={
              Object {
                "avatar_url": "https://avatars3.githubusercontent.com/u/5676?v=4",
                "contributions": 510,
                "html_url": "https://github.com/javache",
                "id": 2,
                "login": "javache",
              }
            }
          />
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Contributors list",
        },
        "ref": null,
        "rendered": "Contributors list",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bxivhb",
            "isStatic": true,
            "rules": Array [
              "
  border-bottom: 1px solid ",
              "#b8b7ad",
              ";
  padding-bottom: 5px;
  text-transform: capitalize;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bxivhb",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ContibutorItem
              contributor={
                Object {
                  "avatar_url": "https://avatars3.githubusercontent.com/u/22032?v=4",
                  "contributions": 623,
                  "html_url": "https://github.com/shergin",
                  "id": 1,
                  "login": "shergin",
                }
              }
            />,
            <ContibutorItem
              contributor={
                Object {
                  "avatar_url": "https://avatars3.githubusercontent.com/u/5676?v=4",
                  "contributions": 510,
                  "html_url": "https://github.com/javache",
                  "id": 2,
                  "login": "javache",
                }
              }
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "function",
            "props": Object {
              "contributor": Object {
                "avatar_url": "https://avatars3.githubusercontent.com/u/22032?v=4",
                "contributions": 623,
                "html_url": "https://github.com/shergin",
                "id": 1,
                "login": "shergin",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "function",
            "props": Object {
              "contributor": Object {
                "avatar_url": "https://avatars3.githubusercontent.com/u/5676?v=4",
                "contributions": 510,
                "html_url": "https://github.com/javache",
                "id": 2,
                "login": "javache",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-ifAKCX",
            "isStatic": true,
            "rules": Array [
              "
  padding-top: 20px;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-ifAKCX",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Contributors list
          </ForwardRef>,
          <ForwardRef>
            <ContibutorItem
              contributor={
                Object {
                  "avatar_url": "https://avatars3.githubusercontent.com/u/22032?v=4",
                  "contributions": 623,
                  "html_url": "https://github.com/shergin",
                  "id": 1,
                  "login": "shergin",
                }
              }
            />
            <ContibutorItem
              contributor={
                Object {
                  "avatar_url": "https://avatars3.githubusercontent.com/u/5676?v=4",
                  "contributions": 510,
                  "html_url": "https://github.com/javache",
                  "id": 2,
                  "login": "javache",
                }
              }
            />
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Contributors list",
          },
          "ref": null,
          "rendered": "Contributors list",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bxivhb",
              "isStatic": true,
              "rules": Array [
                "
  border-bottom: 1px solid ",
                "#b8b7ad",
                ";
  padding-bottom: 5px;
  text-transform: capitalize;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-bxivhb",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ContibutorItem
                contributor={
                  Object {
                    "avatar_url": "https://avatars3.githubusercontent.com/u/22032?v=4",
                    "contributions": 623,
                    "html_url": "https://github.com/shergin",
                    "id": 1,
                    "login": "shergin",
                  }
                }
              />,
              <ContibutorItem
                contributor={
                  Object {
                    "avatar_url": "https://avatars3.githubusercontent.com/u/5676?v=4",
                    "contributions": 510,
                    "html_url": "https://github.com/javache",
                    "id": 2,
                    "login": "javache",
                  }
                }
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "1",
              "nodeType": "function",
              "props": Object {
                "contributor": Object {
                  "avatar_url": "https://avatars3.githubusercontent.com/u/22032?v=4",
                  "contributions": 623,
                  "html_url": "https://github.com/shergin",
                  "id": 1,
                  "login": "shergin",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "2",
              "nodeType": "function",
              "props": Object {
                "contributor": Object {
                  "avatar_url": "https://avatars3.githubusercontent.com/u/5676?v=4",
                  "contributions": 510,
                  "html_url": "https://github.com/javache",
                  "id": 2,
                  "login": "javache",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-ifAKCX",
              "isStatic": true,
              "rules": Array [
                "
  padding-top: 20px;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-ifAKCX",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
